actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,8 +2,14 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
+
+      if template_exists?(method, _prefixes)
```

Comment:
Because of this do we still need `method_for_action` down below?

Alternatively, we might be able to implement this as:

```ruby
def default_head(*)
  logger.info "No template found for #{self.class.name}#{action_name}, rendering head :no_content"
  head :no_content
end

def method_for_action(action_name)
  if method = super 
    method
  elsif template_exists?(action_name.to_s, _prefixes)
    "default_render"
  else
    "default_head"
  end
end
```